packages/mongo-nav/src/mongo-select/MongoSelect.tsx

```diff
@@ -0,0 +1,222 @@
+  data: Array<OrganizationInterface>;
+  current: CurrentOrganizationInterface;
+  constructOrganizationURL: (orgID: string) => string;
+}
+
+const onKeyDown: React.KeyboardEventHandler = e => {
+  if ([keyMap.ArrowUp, keyMap.ArrowDown].includes(e.keyCode)) {
+    e.preventDefault();
+  }
+};
+
+function MongoSelect(
+  props: ProjectMongoSelectProps | OrganizationMongoSelectProps,
+) {
+  const {
+    onClick,
+    variant,
+    onChange,
+    className,
+    urls,
+    isActive = false,
+  } = props;
+
+  let trigger, footer;
+  const { mongoSelect } = urls;
+
+  if (isProject(props)) {
+    trigger = <ProjectTrigger current={props.current} className={className} />;
+    footer = (
+      <li onKeyDown={onKeyDown} role="none" className={projectButtonStyle}>
+        <FocusableMenuItem>
+          <Button href={mongoSelect.viewAllProjects} as="a">
+            View All Projects
+          </Button>
+        </FocusableMenuItem>
+        <FocusableMenuItem>
+          <Button href={mongoSelect.newProject} as="a">
+            + New Project
+          </Button>
+        </FocusableMenuItem>
+      </li>
+    );
+  } else {
+    trigger = (
+      <OrganizationTrigger
+        current={props.current}
+        className={className}
+        urls={urls}
+        isActive={isActive}
+      />
+    );
+    footer = (
+      <MenuItem onKeyDown={onKeyDown} href={mongoSelect?.viewAllOrganizations}>
+        <strong className={viewAllStyle}>View All Organizations</strong>
+      </MenuItem>
+    );
+  }
+
+  const renderProjectOption = (
+    props: ProjectMongoSelectProps,
+    datum: ProjectInterface,
+  ) => {
+    const id = datum.projectId;
+    const content = (
+      <ProjectOption
+        projectName={datum.projectName}
+        href={props.constructProjectURL(datum.orgId, id)}
+      />
+    );
+
+    return (
+      <MenuItem key={id} className={menuItemContainerStyle} onClick={onClick}>
+        {content}
+      </MenuItem>
+    );
+  };
+
+  const renderOrganizationOption = (
+    props: OrganizationMongoSelectProps,
+    datum: OrganizationInterface,
+  ) => {
+    const id = datum.orgId;
+    const content = (
+      <OrganizationOption
+        orgName={datum.orgName}
+        planType={datum.planType}
+        href={props.constructOrganizationURL(id)}
+      />
+    );
+
+    return (
+      <MenuItem key={id} className={menuItemContainerStyle} onClick={onClick}>
+        {content}
+      </MenuItem>
+    );
+  };
+
+  return (
+    <Menu trigger={trigger} className={menuContainerStyle} justify="start">
+      <FocusableMenuItem>
+        <Input onChange={onChange} onKeyDown={onKeyDown} variant={variant} />
+      </FocusableMenuItem>
+
+      <li role="none" className={listContainerStyle}>
+        <ul
+          onKeyDown={e => e.preventDefault()}
+          className={ulContainerStyle}
+          role="menu"
+        >
+          {isProject(props)
+            ? props.data.map(datum => renderProjectOption(props, datum))
```

Comment:
Unless the data is properly typed, Typescript complains about trying to map over the data: ```This expression is not callable.
  Each member of the union type '(<U>(callbackfn: (value: ProjectInterface, index: number, array: ProjectInterface[]) => U, thisArg?: any) => U[]) | (<U>(callbackfn: (value: OrganizationInterface, index: number, array: OrganizationInterface[]) => U, thisArg?: any) => U[])' has signatures, but none of those signatures are compatible with each other.ts(2349)``` So checking the type of data/datum before the map helps to avoid that error